plugins/junit/src/com/intellij/execution/junit/codeInsight/enumsource/EnumSourceProblems.kt

```diff
@@ -0,0 +1,27 @@
+// Copyright 2000-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+package com.intellij.execution.junit.codeInsight.enumsource
+
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.psi.*
+import com.intellij.psi.util.PsiTypesUtil
+
+internal fun checkEnumSourceName(holder: ProblemsHolder,
+                                 enumType: PsiType,
+                                 enumSource: PsiAnnotation,
+                                 name: PsiLiteralExpression) {
+  val modeDefaultValue = findModeDefaultValue(enumSource) ?: return
+  val enumClass = PsiTypesUtil.getPsiClass(enumType) ?: return
+  val enumConstants = enumConstants(enumClass)
+  val supportedMode = annotationContainsIncludeExcludeOrDefaultMode(enumSource.parameterList, modeDefaultValue)
+
+  if (supportedMode && !enumConstants.contains(name.value)) {
+    holder.registerProblem(name, "@EnumSource 'names' contains invalid value. See ${enumClass.name}'s constants.")
+  }
+}
+
+private fun enumConstants(psiClass: PsiClass): Set<String> {
+  return psiClass.fields.asSequence()
+    .filterIsInstance(PsiEnumConstant::class.java)
+    .map { it.name }
+    .toSortedSet()
```

Comment:
forgot to remove sorted because first i had those enums in error msg, than changed my mind to not to have theme in error message.